Nanomips qemu runner

nikolaperic opened the pull request:
Fixed issues when running tests from llvm-test-sute.

Qemu runner script has been added.

Cmake config:

```bash
TOOLCHAIN_DIR="/home/syrmia/llvm-nanomips/nanomips-elf/2022.03-01"
TEST_SUITE_DIR="/home/syrmia/llvm-nanomips/llvm-test-suite"
cmake -G Ninja \
	-DCMAKE_C_COMPILER=$TOOLCHAIN_DIR/bin/clang \
	-C$TEST_SUITE_DIR/cmake/caches/Os.cmake \
	-DTEST_SUITE_RUN_UNDER="$TEST_SUITE_DIR/run-under-qemu.sh $TOOLCHAIN_DIR/bin/qemu-system-nanomips I7200" \
	-DCMAKE_C_FLAGS="--target=nanomips-elf -march=i7200 -mcpu=i7200 -I $TOOLCHAIN_DIR/nanomips-elf/include -I $TOOLCHAIN_DIR/lib/gcc/nanomips-elf/11.2.0/include-fixed -T$TOOLCHAIN_DIR/nanomips-elf/lib/nanomips-r6-soft-tiny/lib/uhi32.ld -Wl,--defsym,__memory_size=256M" \
	-DCMAKE_CXX_COMPILER=$TOOLCHAIN_DIR/bin/clang++ \
	-DCMAKE_CXX_FLAGS="--target=nanomips-elf -march=i7200 -mcpu=i7200 -I $TOOLCHAIN_DIR/nanomips-elf/include -I $TOOLCHAIN_DIR/lib/gcc/nanomips-elf/11.2.0/include-fixed -T$TOOLCHAIN_DIR/nanomips-elf/lib/nanomips-r6-soft-tiny/lib/uhi32.ld -Wl,--defsym,__memory_size=256M" \
	-DTEST_SUITE_COLLECT_CODE_SIZE=OFF \
	-DTEST_SUITE_USE_PERF=ON \
	-DTEST_SUITE_HOST_CC=gcc \
	-DTEST_SUITE_USER_MODE_EMULATION=ON \
	-DTEST_SUITE_RUN_BENCHMARKS=ON \
	-DTEST_SUITE_COLLECT_COMPILE_TIME=OFF \
	-DTEST_SUITE_SUBDIRS="$TEST_SUITE_DIR/SingleSource;$TEST_SUITE_DIR/MultiSource" \
	$TEST_SUITE_DIR
```
Run tests:
```bash
$ cd <llvm_test_suite_build_dir> && <llvm_build_dir>/bin/llvm-lit -v --no-progress-bar -o results.json .
```

Commit:
Fix llvm-size command line argument

Commit:
Fix issue with setting ldflags

The spaces within the strings added to LDFLAGS were escaped using "\".
This caused unknown flag errors. Appending them individually fixes this issue.

Commit:
Set target format for strip tool

Commit:
Add qemu runner script

farazs-github commented on run-under-qemu.sh:
```diff
@@ -0,0 +1,25 @@
+shift 2
+
+if [ $# -eq 0 ]; then
+    $qemu_path -cpu I7200 -semihosting -nographic -kernel $test_name 2> /dev/null
```
For best results, use `-m 256' and also adjust __memory_size in C/CXX flags to match this. Should reduce the number of failures you see!

farazs-github commented on run-under-qemu.sh:
```diff
@@ -0,0 +1,25 @@
+# Invocation run_under_qemu.sh [qemu_path] [test_name] [test_args...]
+
+if [ $# -lt 2 ]; then
+    echo "Usage: run_under_qemu.sh [qemu_path] [test_name] [test_args...]"
```
Can we make this more generic, by allowing some of the qemu-args to be passed in as well so that it can work for more targets? For one, the ability to parametrize `-cpu' would be useful in testing different configurations.

Commit:
Qemu runner script improvements

- Add the ability to specify CPU through command line arguments
- Increase test stack memory
